### ISSUE TITLE
Failed to load pretrained DINO to reproduce training

### ISSUE THREAD

**Issue by oggyfaker** [SebastianJanampa/DT-LSD#6]
Hi author, i have some issue when use pretrained [DINO Repo](https://github.com/IDEA-Research/DINO/tree/main) for training from scratch. I want to use **only pretrained DINO (not resume the weight you trained)** for training new dataset with this bash. However the backbone from DINO cant load checkpoint. Can you point me out the way when you train your model from scratch ?  and if i want to train model on new dataset , what should i do to make sure the checkpoints is matched with model ? 

Here is the pretrained i downloaded from [DINO Google Drive](https://drive.google.com/drive/folders/1qD5m1NmK0kjE5hh-G17XUX751WsEG-h_)

![Image](https://github.com/user-attachments/assets/b787fafe-fae5-4e36-81de-8103ba08b3cc)

This is my bash , i remove `resume`

```bash 
 python main.py \
	--output_dir logs/SWIN-MS4-24-epochs -c config/DTLSD/DTLSD_4scale_swin.py --coco_path data/wireframe_processed \
	--no_opt\
	--pretrain\
	--pretrain_model_path pretrain/checkpoint0033_4scale.pth\
	--options dn_number=300 embed_init_tgt=TRUE \
	dn_label_coef=1.0 dn_bbox_coef=1.0 use_ema=False \
	dn_box_noise_scale=1.0 epochs=24 lr_drop=20 \
	num_classes=2 dn_labelbook_size=2 focal_alpha=0.1 batch_size=2
``` 

And this is my bug log 

<img width="1441" alt="Image" src="https://github.com/user-attachments/assets/e14bba3c-2280-48b0-861b-67c3b6225a68" />

Hope to see yours answer soon !

**Comment by SebastianJanampa** [comment]
Please use this command
``` shell
 python main.py \
	--output_dir logs/SWIN-MS4-24-epochs -c config/DTLSD/DTLSD_4scale_swin.py --coco_path data/wireframe_processed \
	--no_opt\
	--pretrain\
	--resume pretrain/checkpoint0033_4scale.pth\
	--options dn_number=300 embed_init_tgt=TRUE \
	dn_label_coef=1.0 dn_bbox_coef=1.0 use_ema=False \
	dn_box_noise_scale=1.0 epochs=24 lr_drop=20 \
	num_classes=2 dn_labelbook_size=2 focal_alpha=0.1 batch_size=2
```

The argument `pretrain_model_path` is used in DINO to train DINO on the COCO dataset after being trained first on the OBJECT365 dataset.

**Comment by oggyfaker** [comment]
After use your command , i got this warning, some parameter of layer in model will be in initialized with new weight? I see all the parameter of decoder blocks and box_embed are not existed in current model , the performance if i training on new dataset will be bad ? 

Any recommendation for config parameter (learning rate , scheduler step ) , model  if i add new dataset with more than 2 classes ? 

```
transformer.decoder.bbox_embed.1.layers.0.weight NOT appear in current model.  
transformer.decoder.bbox_embed.1.layers.0.bias NOT appear in current model.  
transformer.decoder.bbox_embed.1.layers.1.weight NOT appear in current model.  
transformer.decoder.bbox_embed.1.layers.1.bias NOT appear in current model.  
transformer.decoder.bbox_embed.1.layers.2.weight NOT appear in current model.  
transformer.decoder.bbox_embed.1.layers.2.bias NOT appear in current model.  
transformer.decoder.bbox_embed.2.layers.0.weight NOT appear in current model.  
transformer.decoder.bbox_embed.2.layers.0.bias NOT appear in current model.  
transformer.decoder.bbox_embed.2.layers.1.weight NOT appear in current model.  
transformer.decoder.bbox_embed.2.layers.1.bias NOT appear in current model.  
transformer.decoder.bbox_embed.2.layers.2.weight NOT appear in current model.  
transformer.decoder.bbox_embed.2.layers.2.bias NOT appear in current model.  
transformer.decoder.bbox_embed.3.layers.0.weight NOT appear in current model.  
transformer.decoder.bbox_embed.3.layers.0.bias NOT appear in current model.  
transformer.decoder.bbox_embed.3.layers.1.weight NOT appear in current model.  
transformer.decoder.bbox_embed.3.layers.1.bias NOT appear in current model.  
transformer.decoder.bbox_embed.3.layers.2.weight NOT appear in current model.  
transformer.decoder.bbox_embed.3.layers.2.bias NOT appear in current model.  
transformer.decoder.bbox_embed.4.layers.0.weight NOT appear in current model.  
transformer.decoder.bbox_embed.4.layers.0.bias NOT appear in current model.  
transformer.decoder.bbox_embed.4.layers.1.weight NOT appear in current model.  
transformer.decoder.bbox_embed.4.layers.1.bias NOT appear in current model.  
transformer.decoder.bbox_embed.4.layers.2.weight NOT appear in current model.  
transformer.decoder.bbox_embed.4.layers.2.bias NOT appear in current model.  
transformer.decoder.bbox_embed.5.layers.0.weight NOT appear in current model.  
transformer.decoder.bbox_embed.5.layers.0.bias NOT appear in current model.  
transformer.decoder.bbox_embed.5.layers.1.weight NOT appear in current model.  
transformer.decoder.bbox_embed.5.layers.1.bias NOT appear in current model.  
transformer.decoder.bbox_embed.5.layers.2.weight NOT appear in current model.  
transformer.decoder.bbox_embed.5.layers.2.bias NOT appear in current model.  
bbox_embed.0.layers.0.weight NOT appear in current model.  
bbox_embed.0.layers.0.bias NOT appear in current model.  
bbox_embed.0.layers.1.weight NOT appear in current model.  
bbox_embed.0.layers.1.bias NOT appear in current model.  
bbox_embed.0.layers.2.weight NOT appear in current model.  
bbox_embed.0.layers.2.bias NOT appear in current model.  
bbox_embed.1.layers.0.weight NOT appear in current model.  
bbox_embed.1.layers.0.bias NOT appear in current model.  
bbox_embed.1.layers.1.weight NOT appear in current model.  
bbox_embed.1.layers.1.bias NOT appear in current model.  
bbox_embed.1.layers.2.weight NOT appear in current model.  
bbox_embed.1.layers.2.bias NOT appear in current model.  
bbox_embed.2.layers.0.weight NOT appear in current model.  
bbox_embed.2.layers.0.bias NOT appear in current model.  
bbox_embed.2.layers.1.weight NOT appear in current model.  
bbox_embed.2.layers.1.bias NOT appear in current model.  
bbox_embed.2.layers.2.weight NOT appear in current model.  
bbox_embed.2.layers.2.bias NOT appear in current model.  
bbox_embed.3.layers.0.weight NOT appear in current model.  
bbox_embed.3.layers.0.bias NOT appear in current model.  
bbox_embed.3.layers.1.weight NOT appear in current model.  
bbox_embed.3.layers.1.bias NOT appear in current model.  
bbox_embed.3.layers.2.weight NOT appear in current model.  
bbox_embed.3.layers.2.bias NOT appear in current model.  
bbox_embed.4.layers.0.weight NOT appear in current model.  
bbox_embed.4.layers.0.bias NOT appear in current model.  
bbox_embed.4.layers.1.weight NOT appear in current model.  
bbox_embed.4.layers.1.bias NOT appear in current model.  
bbox_embed.4.layers.2.weight NOT appear in current model.  
bbox_embed.4.layers.2.bias NOT appear in current model.  
bbox_embed.5.layers.0.weight NOT appear in current model.  
bbox_embed.5.layers.0.bias NOT appear in current model.  
bbox_embed.5.layers.1.weight NOT appear in current model.  
bbox_embed.5.layers.1.bias NOT appear in current model.  
bbox_embed.5.layers.2.weight NOT appear in current model.  
bbox_embed.5.layers.2.bias NOT appear in current model.  
transformer.decoder.class_embed.0.weight NEW parameter.  
transformer.decoder.class_embed.0.bias NEW parameter.  
transformer.enc_out_class_embed.weight NEW parameter.  
transformer.enc_out_class_embed.bias NEW parameter.  
label_enc.weight NEW parameter.  
```

**Comment by SebastianJanampa** [comment]
Hi,
That is an expected warning. We use the DINO weights to initialize the backbone and encoder, not the decoder. We need the pretrained weights, especially because DT-LSD uses the SWIN backbone.

The `bbox_embed` and `class_embed`  are initialized here
https://github.com/SebastianJanampa/DT-LSD/blob/3592ce27153452dd3241d483f68169dfd9433b11/models/dtlsd/dtlsd.py#L110-L135

We do not initialize the decoder with the DINO weights because, in DINO, those layers estimate the bounding box as `xywh,` while DT-LSD estimates the endpoints of a line. On the other hand, using DINO weights for the backbone and encoder is beneficial because they extract and enhance feature maps. 

Now, about training on a different dataset. From what I understand, you want to estimate the lines' endpoints and classify them into different groups. Am i right?  I would recommend keeping the learning rate and scheduler as they are. Just change this line in the config file https://github.com/SebastianJanampa/DT-LSD/blob/3592ce27153452dd3241d483f68169dfd9433b11/config/DTLSD/DTLSD_4scale_swin.py#L3 
Use `3`  (num_classes +1) instead of `2`. We use the +1to refer that the detected object has no class.

I hope this answer helps you.
Good luck with your project

**Comment by oggyfaker** [comment]
Thanks for your response !
